include/exadg/fluid_structure_interaction/precice/driver_fluid.h

```diff
@@ -218,7 +217,36 @@ class DriverFluid : public Driver<dim, Number>
 
       // post-solve
       if(is_new_time_window)
+      {
+        // computes new time-step size
         fluid->time_integrator->advance_one_timestep_post_solve();
+        // The current implementation does not allow to synchronize the time step size between the
+        // fluid field and precice. Instead, the fluid field dictates the time step size. There are
+        // two reasons behind that: first, in implicit (strong) coupling schemes, we need matching
+        // time-window sizes as the time-step size push back of the BDF scheme happens in the
+        // 'advance_one_timestep_post_solve()' and we cannot change two subsequent time-step sizes
+        // without a push back operation, otherwise we falsify the time integrator (this would still
+        // work with explicit (weak) coupling schemes). Second, the treatment of the boundary
+        // conditions is not handled correctly when using subcycling, as we would apply the 'whole
+        // displacement' within the first time-step. In order to cope with subcycling, either the
+        // boundary condition needs to be adapted or we need to wait for a newer preCICE version
+        // (currently v2.3.0) which can handle this. Hence, we do not adjust the time-step-size of
+        // the fluid solver, but rather Assert that it is still valid. Note that we would otherwise
+        // run into a preCICE assert reporting the same issue.
+        // fluid->time_integrator->set_current_time_step_size(
+        //   std::min(this->time_until_next_coupling_window,
+        //   fluid->time_integrator->get_time_step_size()));
+        // Have a look in the base class for more explanations.
```

Comment:
This comment on base class is not clear to me? Is it needed? I would remove the code above.